File: database/models/MedicalRecord/Prescription.js
import { Schema, model } from 'mongoose';

const prescriptionSchema = new Schema({
    userId: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    doctorId: {
        type: Schema.Types.ObjectId,
        ref: 'Doctor',
    },
    diagnosis: {//chẩn đoán
        type: String,
        required: true
    },
    medicines: [{
        name: {
            type: String,
            required: true
        },
        dosage: {
            type: String,
            required: true
        },
        frequency: {
            type: String,
            required: true
        },
        duration: {
            type: String,
            required: true
        },
        notes: String
    }],
    notes: String,
    date: {
        type: Date,
        default: Date.now
    }
}, {
    timestamps: true
});

export default model('Prescription', prescriptionSchema); 

File: mainflutter/lib/models/prescription.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'prescription.freezed.dart';
part 'prescription.g.dart';

@freezed
class Medicine with _$Medicine {
  const factory Medicine({
    required String name,
    required String dosage,
    required String frequency,
    required String duration,
    String? notes,
  }) = _Medicine;

  factory Medicine.fromJson(Map<String, dynamic> json) => _$MedicineFromJson(json);
}

@freezed
class Prescription with _$Prescription {
  const factory Prescription({
    String? id,
    String? userId,
    String? doctorId,
    required String diagnosis,
    required List<Medicine> medicines,
    String? notes,
    required DateTime date,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) = _Prescription;

  factory Prescription.fromJson(Map<String, dynamic> json) => _$PrescriptionFromJson(json);
} 